### ISSUE TITLE
ubuntu bionic docker image version?

### ISSUE THREAD

**Issue by petenorth** [Azure/azure-functions-docker#355]
There are amd64, arm64 and arm32 architecture bionic runtime dependencies images - it seems odd that only the arm32 architecture image is used to create an Azure Functions 3.0-bionic-arm32v7 why not use the amd64 architecture bionic runtime dependencies image to create a 3.0-bionic image (with amd64 architecture).

Our image scanning (Aquasec / Trivy) reports zero high/critical vulnerabilities for this base image.

**Comment by balag0** [comment]
@ahmelsayed I think we have always used the debian images, do you remember if we tried using the bionic ubuntu amd64 base image?

**Comment by ahmelsayed** [comment]
Honestly I only added the arm32 bionic image because someone (azure-iot team) asked for it specifically. I have no issues with adding other archs for bionic, there was just never an ask for it until now. We just use debian for our default images.

**Comment by divyagandhii** [comment]
@balag0 have you started working on this? Else I can pick this up.

**Comment by balag0** [comment]
No I haven't. Feel free to assign it to yourself. thanks.

**Comment by divyagandhii** [comment]
Sounds good. Thanks

**Comment by paulgear** [comment]
Any update on this?  I'd really like to have the option of running on Ubuntu for consistency with mcr.microsoft.com/dotnet/aspnet and mcr.microsoft.com/dotnet/sdk, so that we can consolidate our software supply chain on a smaller base, and maximise the overlap in our docker layers for performance.

**Comment by paulgear** [comment]
Looking at the source tree it appears the Dockerfiles and build scripts are maintained manually, rather than through some meta-build system.  If I've understood it correctly, I'm happy to have a crack at a PR to add the images we need.

**Comment by ankit-h** [comment]
Any update on this? the latest v3 images has couple of vulnerabilities and was hoping to have an updated image with the fixes

**Comment by balag0** [comment]
@paulgear We are working on refactoring some of the images, so planning on working on this right after. But hopefully you can unblock by building a custom image based on the required bionic images until then. Does that work?

**Comment by paulgear** [comment]
Hi @balag0, that's fine, I just wouldn't like to waste a lot of time creating build infrastructure which is not going to be useful.  Do you have any public info about what refactoring is planned?

**Comment by balag0** [comment]
There are 2 workitems planned
step1 is to extract out some common base layers across images and build them once and reuse.
step2 is to generate dockerfiles using templates instead of having to edit the docker files directly.

It might be a while before step2 is started, but step1 is in progress currently. I think it would be helpful if you can share more details on the build infrastructure you are considering. May be we can use that for step 2 :)

**Comment by paulgear** [comment]
@balag0 Any chance you can take a look at PR #436?  It's building OK for me in our work AzDO instance.

**Comment by vivekjilla** [comment]
Ubuntu 18.04 (bionic) is end of life, and the images use Debian bases. Published images are refreshed regularly, so CVE fixes are picked up over time. With no activity since April 28, 2021 (about 5 years), we're closing this. If you need CVE coverage or a specific base on a current supported tag/digest, please open a new issue.
